packages/tupaia-web/src/views/ProjectSelectModal.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2024 Beyond Essential Systems Pty Ltd
+ */
+import React from 'react';
+import styled from 'styled-components';
+import { useNavigate } from 'react-router-dom';
+import { ProjectSelectForm } from '@tupaia/ui-components';
+import { useProjects, useUser } from '../api/queries';
+import { useEditUser } from '../api/mutations';
+import { Modal } from '../components';
+import { useModal } from '../utils';
+import { MODAL_ROUTES } from '../constants';
+
+const ModalBody = styled.div`
+  text-align: left;
+  padding: 1.2rem;
+  width: 46rem;
+  max-width: 100%;
+  margin-top: -1rem;
+
+  > div {
+    max-block-size: 28rem;
+  }
+
+  .MuiDialogActions-root {
+    position: relative;
+    top: 1rem;
+  }
+`;
+
+export const ProjectSelectModal = () => {
+  const { data } = useUser();
+  const projectId = data?.project?.id;
+  const { data: projects, isFetching } = useProjects();
+  const { closeModal } = useModal();
+  const navigate = useNavigate();
+
+  const onSelectProject = data => {
```

Comment:
What are your thoughts on keep the project select as a link and having a `useEffect` somewhere to set the user preferences on project change (which I think is a bit more like Datatrak Web)? Mainly because I think links should always be links, not buttons that then navigate somewhere